src/encoded/tests/test_static_page.py

```diff
@@ -60,87 +62,117 @@ def help_page_json_deleted():
             "header-depth": 4,
             "list-styles": ["decimal", "lower-alpha", "lower-roman"]
         },
-        "status" : "deleted"
+        "status": "deleted"
     }
 
 
-@pytest.fixture(scope='module')
-def posted_help_page_section(testapp, help_page_section_json):
-    try:
-        res = testapp.post_json('/static-sections/', help_page_section_json, status=201)
-        val = res.json['@graph'][0]
-    except webtest.AppError:
-        res = testapp.get('/' + help_page_section_json['uuid'], status=301).follow()
-        val = res.json
-    return val
+# Move this to some test utilities place when debugged
+JSON_HEADERS = {'Content-Type': 'application/json', 'Accept': 'application/json'}
 
 
-@pytest.fixture(scope='module')
-def help_page(testapp, posted_help_page_section, help_page_json):
+# Move this to some test utilities place when debugged
+@contextlib.contextmanager
+def posted_page(testapp, base_url, content):
+    uuid = content.get('uuid')
     try:
-        res = testapp.post_json('/pages/', help_page_json, status=201)
-        val = res.json['@graph'][0]
+        [val] = testapp.post_json(base_url, content, status=(201, 301)).maybe_follow().json['@graph']
     except webtest.AppError:
-        res = testapp.get('/' + help_page_json['uuid'], status=301).follow()
-        val = res.json
-    return val
+        val = testapp.get(url_path_join('/', base_url, uuid),
+                          headers=JSON_HEADERS,
+                          status=(200, 301)).maybe_follow().json
+    uuid = uuid or val.get('uuid')
+    yield val
+    if uuid:
+        # Note: Without JSON headers, this will get a 415 even if it's not using the result data. -kmp 23-Feb-2021
+        testapp.delete(url_path_join('/', uuid), headers=JSON_HEADERS)
 
 
-@pytest.fixture(scope='module')
-def help_page_deleted(testapp, posted_help_page_section, help_page_json_draft):
```

Comment:
Note that these fixtures were kind of a mess, with the draft json going into the deleted page, etc. I tried to clean that kind of thing up. It may be worth looking at the new file in direct mode, not side-by-side.